framework/src/modules/sequence/sequence_module.ts

```diff
@@ -29,4 +32,54 @@ export class SequenceModule extends BaseModule {
 			balance: BigInt(0),
 		},
 	};
+
+	public async beforeTransactionApply(input: TransactionApplyInput): Promise<void> {
+		const { tx, stateStore } = input;
+		const senderAddress = getAddressFromPublicKey(tx.senderPublicKey);
+		const senderAccount = await stateStore.account.getOrDefault<{ sequence: { nonce: bigint } }>(
+			senderAddress,
+		);
+
+		// Throw error when tx nonce is lower than the account nonce
+		if (tx.nonce < senderAccount.sequence.nonce) {
+			throw new SequenceModuleError(
```

Comment:
maybe pass the data to `SequenceModuleError` and build the message within it?

Reply:
Removed this error and using existing `NonceOutOfBoundsError` error instead